### PR TITLE
Add docs for the CSRF prevention plugin

### PR DESCRIPTION
**Pull request by SimonSapin**
Fixes https://github.com/apollographql/router/issues/1039

The plugin was added in https://github.com/apollographql/router/pull/1006

Docs are imported from [Apollo Server’s own docs](https://www.apollographql.com/docs/apollo-server/security/cors/#preventing-cross-site-request-forgery-csrf) and adapted.

### CODE REVIEW

File: docs/source/configuration/csrf.mdx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+- There is a `X-Apollo-Operation-Name` header. This header is sent with all operations (`POST` or `GET`) by [Apollo iOS](/ios) (v0.13.0+) and [Apollo Kotlin](/kotlin) (all versions, including its former name "Apollo Android"), so any request from Apollo iOS or Apollo Kotlin will be executed.
+- There is a `Apollo-Require-Preflight` header.
+
+Note that all HTTP header names are case-insensitive.
+
+> CSRF prevention is only applied to requests that will execute GraphQL operations, not to requests that would load [landing pages](../api/plugin/landing-pages) or run [health checks](../monitoring/health-checks).)
```
Comment (SimonSapin) [inline docs/source/configuration/csrf.mdx:28]: Does the router have a landing page, health checks, or anything else that’s not GraphQL requests? Do these go through plugins?

Comment (SimonSapin) [inline docs/source/configuration/csrf.mdx:28]: This probably needs change, at least for fixing the links.

Does the router have a landing page, health checks, or anything else that’s not GraphQL requests? Do these go through plugins?

Comment (Geal) [reply docs/source/configuration/csrf.mdx:28]: Yes, we have other routes. Looking at this code, for health check, landing page and graphql endpoint the request first goes through CORS, then plugins. While for plugin specific endpoint, the request goes through directly:
https://github.com/apollographql/router/blob/f7fc365d921be3503297caa9810d976d897e9571/apollo-router/src/axum_http_server_factory.rs#L101-L132

**Comment by abernix** [comment]
This closes #1039, right?

**Comment by SimonSapin** [comment]
Oh indeed. I’ve added the magic keyword in the PR description.

**Review by o0Ignition0o — APPROVED** [review]
This looks great, congratulations! 🎉

**Pull request merged**